### ISSUE TITLE
Do not retry when network is not yet up in booting

### ISSUE THREAD

**Issue by EaseTheWorld** [square/okhttp#4387]
My application is using retrofit2 + okhttp3 on redhat6 workstation.
I wanted to run at booting so added in crontab @reboot
But it seemed my app runs before network is up.
I expected the retry logic could recover after a few seconds but it never even retried.
I digged into recover() in RetryAndFollowUpInterceptor.java but streamAllocation.hasMoreRoutes() return false.
Any solution for this? or do I have to manually retry until network is up?

**Comment by swankjesse** [comment]
Yep. OkHttp has no idea whether the network will ever come up. If it _did_ try to hold requests until the network came up a bad network would trigger a memory leak, and could crash the process.

**Comment by EaseTheWorld** [comment]
Thank you for quick response.
I'm using retrofit2 enqueue(), then what would be the safest retry?
Best I can think of is overriding intercept and repeat like
`for(i<retryCount) { try { chain.proceed() } catch (java.net.ConnectException)}`
Any suggestion would be appreciated.

**Comment by EaseTheWorld** [comment]
I found similar solution at
https://stackoverflow.com/questions/24562716/how-to-retry-http-requests-with-okhttp-retrofit
